src/main/java/de/unistuttgart/iste/gits/reward/service/RewardService.java

```diff
@@ -136,7 +134,13 @@ public RewardScores calculateScoresOnContentWorkedOn(UserProgressLogEvent event)
     public AllRewardScoresEntity initializeRewardScores(UUID courseId, UUID userId) {
         AllRewardScoresEntity allRewardScoresEntity = new AllRewardScoresEntity();
         allRewardScoresEntity.setId(new AllRewardScoresEntity.PrimaryKey(courseId, userId));
-        allRewardScoresEntity.setHealth(initializeRewardScoreEntity(100));
+
```

Comment:
the following two lines invoke the course service and the content service. This function will crash, if one of those is not available. I would recommend wrapping it in a try-catch and catching the corresponding exceptions (CourseClientException and ContentClientException). If they occur, I would then fall back to 100 as default value.